docs/api/latest.md

```diff
@@ -1553,6 +1553,496 @@ It is possible to supply a custom error message by specifying a `message` proper
+    }
+  ]
+}
+```
+
+#### Filename clashes
+
+If the filename of an file being uploaded is the same as an existing file, the new file will have it's name changed by adding the current timestamp:
+
+* Existing filename: `my-image.jpg`
+* New filename: `test-1480482847099.jpg`
+
+### Referencing files from another collection
+
+Once a file is uploaded, it's identifier can be used to create a reference from another collection. For this example we have a collection called `books` with the following schema:
+
+```json
+{
+  "fields": {
+    "title": {
+      "type": "String",
+      "required": true
+    },
+    "content": {
+      "type": "String",
+      "required": true
+    },
+    "image": {
+      "type": "Reference",
+      "settings": {
+        "collection": "mediaStore"
+      }
+    }
+  },
+  "settings": {
+    "cache": true,
+    "count": 40,
+    "compose": true,
+    "sort": "title",
+    "sortOrder": 1
+  }
+}
+```
+
+The `image` field is a Reference field which will lookup the default `mediaStore` collection to resolve the reference. Having uploaded an image file and received it's metadata, we can now send a POST to the `books` collection with the image identifier.
+
+```http
+POST /1.0/library/books HTTP/1.1
+Host: api.somedomain.tech
+Content-Type: application/json
+Authorization: Bearer 8df4a823-1e1e-4bc4-800c-97bb480ccbbe
+
+{
+  "title": "Harry Potter and the Philosopher's Stone",
+  "content": "Harry Potter and the Philosopher's Stone is the first novel in the Harry Potter series and J. K. Rowling's debut novel, first published in 1997 by Bloomsbury.",
+  "image": "58576e72bafa53b625aebd4f"
+}
+```
+
+A subsequent GET request for this book would return a response such as:
+
+```json
+{
+  "title": "Harry Potter and the Philosopher's Stone",
+  "content": "Harry Potter and the Philosopher's Stone is the first novel in the Harry Potter series and J. K. Rowling's debut novel, first published in 1997 by Bloomsbury.",
+  "image": {
+    "fileName":"my-image.jpg",
+    "mimetype":"image/jpeg",
+    "width":1920,
+    "height":1080,
+    "path":"workspace/media/2016/12/19/my-image.jpg",
+    "contentLength":173685,
+    "awsUrl":"https://bucketName.s3.amazonaws.com/workspace/media/2016/12/19/my-image.jpg",
+    "_createdAt":1482124902978,
+    "_createdBy":"your-client-key",
+    "_version":1,
+    "_id":"58576e72bafa53b625aebd4f"
+  }
+}
+```
+
+### Pre-signed URLs
+
+Pre-signed URLs are useful to allow your users or applications to be able to upload a file without requiring an access token. When you request a pre-signed URL, you must provide an access token (see [Authentication](#x)) and specify an expected filename and MIME type for the file to be uploaded. The pre-signed URLs are valid only for the duration specified in the main configuration file or passed in the request for the signed URL.
+
+
+#### Configuration
+
+```json
+"media": {
+  "enabled": true,
+  "tokenSecret": "catbus-goat-omelette",
+  "tokenExpiresIn": "10h"
```

Comment:
I didn't know we supported this human-friendly format for specifying the TTL. We expand on this in the documentation, perhaps?